packages/pieces/community/common/src/lib/helpers/index.ts

```diff
@@ -111,12 +111,15 @@ export function createCustomApiCallAction({ auth, baseUrl, authMapping, descript
           }
         }
       }
-
+      let parsedQueryparam = queryParams
+      if(typeof queryParams === 'string'){
+        parsedQueryparam = JSON.parse(queryParams)
+      }
```

Comment:
Hello there, there could be a chance someone passes an invalid JSON, in that case we probably should throw an error saying "Query params are not a valid json" with the stack error so people can debug their query params.